xbmc/addons/IAddon.h

```diff
@@ -92,6 +92,7 @@ namespace ADDON
     virtual bool GetSettingString(const std::string& key, std::string& value) = 0;
     virtual CAddonSettings* GetSettings() const =0;
     virtual const ADDONDEPS &GetDeps() const =0;
+    virtual int NoDeps(std::map<std::string,std::string>& added) const = 0;
```

Comment:
same here e.g. `GetMissingDependencies` and if we only need the map for the recursive call, we should think about to add a convenient method without the parameter and init the map there.

```
virtual int GetMissingDependencies() const {
  std::map<std::string,std::string>& addons;
  return GetMissingDependencies(addons);
}